### PR TITLE
M-64: Added question form, has-many component, room editing and question deletion

### PR DESCRIPTION
**Pull request by gustass5**
https://trello.com/c/MRZCYNo3/64-implement-question-form
https://trello.com/c/vCIaTBAK/76-implement-room-editing
https://trello.com/c/UuvGcLZe/77-implement-room-question-deletion

### CODE REVIEW

File: app/Http/Controllers/EditRoomController.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            if(!$questionsIds->contains($question->id)){
+                $room->questions()->find($question->id)->delete();
+                // [TODO]: Also delete question answers here;
+            }
+        });
+
+        collect($request['questions'])->each(function($question) use ($room){
+            $room->questions()->updateOrCreate(["id" => $question["id"]], ["question" => $question['value']]);
+        });
+        return Inertia::render("Rooms/EditPage", ["room" => $room, "questions" => $room->questions()->get()]);
```
Comment (Luksys3) [inline app/Http/Controllers/EditRoomController.php:35]: You should use redirect

File: resources/js/Pages/Rooms/EditPage.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+					}}
+				/>
+			</RoomForm>
+		</Panel>
+	);
+};
+
+(EditPage as any).layout = (page: ReactElement) => {
+	return (
+		<AuthenticatedLayout title="Create new room">
```
Comment (Luksys3) [inline resources/js/Pages/Rooms/EditPage.tsx:82]: Edit?

File: app/Http/Controllers/EditRoomController.php (hunk 1 above)
Comment (gustass5) [reply app/Http/Controllers/EditRoomController.php:35]: I will fix it in my next PR

File: resources/js/Pages/Rooms/EditPage.tsx (hunk 2 above)
Comment (gustass5) [reply resources/js/Pages/Rooms/EditPage.tsx:82]: Same, probably it would be the best if I just fix it in my next PR

**Pull request merged**